### PR TITLE
[tock] Add Tock UART tests.

### PR DESCRIPTION
**Pull request by jrvanwhy**
This test the ability of a Tock app to perform console writes and reads over the UART. It is an integration tests that tests the Console syscall driver, the UART virtualizer, and the UART peripheral driver.

### CODE REVIEW

**Comment by jrvanwhy** [comment]
I think the CI error is a true positive, debugging now.

File: hw/top_earlgrey/BUILD [hunk 1]
```diff
@@ -129,6 +129,26 @@ fpga_cw310(
     rsa_key = {"//sw/device/silicon_creator/rom_ext/keys/fake:rom_ext_test_private_key_0": "test_key_0"},
 )
 
+fpga_cw310(
+    name = "fpga_hyper310_test_rom",
```
Comment (a-will) [inline hw/top_earlgrey/BUILD:133]: Is there a compelling reason to add a test ROM environment for hyper310 and not just use `//hw/top_earlgrey:fpga_cw310_sival_rom_ext` or `//hw/top_earlgrey:fpga_hyper310_rom_ext`?

If not, it'd be nice if we didn't expand the maintenance surface, and the changes to this file and rules/opentitan/defs.bzl can be removed.

File: sw/device/silicon_owner/tock/tests/uart/BUILD [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    apps = [":tab"],
+    kernel = "//sw/device/silicon_owner/tock/kernel",
+)
+
+opentitan_test(
+    name = "uart_test",
+    exec_env = {
+        "//hw/top_earlgrey:fpga_hyper310_test_rom": None,
+    },
+    hyper310 = cw310_params(
```
Comment (a-will) [inline sw/device/silicon_owner/tock/tests/uart/BUILD:43]: Consider using one of the existing environments:

```suggestion
    exec_env = {
        "//hw/top_earlgrey:fpga_cw310_sival_rom_ext": None,
        "//hw/top_earlgrey:fpga_hyper310_rom_ext": "cw310",
    },
    cw310 = cw310_params(
```

File: hw/top_earlgrey/BUILD (hunk 1 above)
Comment (jrvanwhy) [reply hw/top_earlgrey/BUILD:133]: Short answer: no.

Long answer: I was originally seeing test failures on cw310 that I believed were a result of a cw310 issue, so I moved this test over to hyper310. However, the issues eventually showed up on hyper310 as well and they were traced to a Tock driver bug. Now that the bug is fixed, we can switch back to cw310.

I don't think Tock will run with a ROM EXT yet, so those particular environments aren't an option.

**Pull request closed without merging**